pokemongo_bot/walkers/step_walker.py

```diff
@@ -22,9 +23,13 @@ def __init__(self, bot, dest_lat, dest_lng, dest_alt=None, precision=0.5):
             self.dest_alt = dest_alt
 
         self.saved_location = None
+        self.last_update = 0
 
     def step(self, speed=None):
-        sleep(1)
+        now = time.time()
+
+        sleep(1 - min(now - self.last_update, 1))
```

Comment:
abs(now - self.last_update)

And you do want to take the max not the min!

this should be absolute...  sometimes time can tick back (especially in in while True... loops)

Then you'll get In 
```
[18]: time.sleep(-0.00001)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-18-8e671aa2b61c> in <module>()
----> 1 time.sleep(-0.00001)

ValueError: sleep length must be non-negative

In [19]: 
```